src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -478,32 +482,43 @@ private void InitializePackagesFolderInfo(string packagesFolderPath, bool exclud
                 throw new ArgumentNullException(nameof(secondarySources));
             }
 
-            if (nuGetProject is INuGetIntegratedProject)
-            {
-                // project.json based projects are handled here
-                return await PreviewUpdatePackagesForBuildIntegratedAsync(
-                    packageId,
-                    packageIdentities,
-                    nuGetProject,
-                    resolutionContext,
-                    nuGetProjectContext,
-                    primarySources,
-                    secondarySources,
-                    token);
-            }
-            else
+            var tasks = new List<Task<IEnumerable<NuGetProjectAction>>>();
+
+            foreach (var project in nuGetProjects)
             {
-                // otherwise classic style packages.config style projects are handled here
-                return await PreviewUpdatePackagesForClassicAsync(
-                    packageId,
-                    packageIdentities,
-                    nuGetProject,
-                    resolutionContext,
-                    nuGetProjectContext,
-                    primarySources,
-                    secondarySources,
-                    token);
+                if (project is INuGetIntegratedProject)
+                {
+                    // project.json based projects are handled here
+                    tasks.Add(Task.Run(async ()
+                        => await PreviewUpdatePackagesForBuildIntegratedAsync(
+                            packageId,
+                            packageIdentities,
+                            project,
+                            resolutionContext,
+                            nuGetProjectContext,
+                            primarySources,
+                            secondarySources,
+                            token)));
+                }
+                else
+                {
+                    // otherwise classic style packages.config style projects are handled here
+                    tasks.Add(Task.Run(async ()
```

Comment:
This change for uwp should not change PC to run in parallel. I'm not sure that the gather caching can handle this, it may result in duplicate http calls. Also there is zero throttling here